src/ResourceManager/Insights/Commands.Insights/Metrics/GetAzureRmMetricCommand.cs

```diff
@@ -37,33 +44,35 @@ public class GetAzureRmMetricCommand : InsightsClientCmdletBase
         /// <summary>
         /// Gets or sets the ResourceId parameter of the cmdlet
         /// </summary>
-        [Parameter(Position = 0, Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "The resource Id")]
+        [Parameter(ParameterSetName = GetAzureRmAMetricParamGroup, Position = 0, Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "The resource Id")]
+        [Parameter(ParameterSetName = GetAzureRmAMetricFullParamGroup, Position = 0, Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "The resource Id")]
         [ValidateNotNullOrEmpty]
         public string ResourceId { get; set; }
 
         /// <summary>
-        /// Gets or sets the timegrain parameter of the cmdlet
+        /// Gets or sets the aggregation type parameter of the cmdlet
         /// </summary>
-        [Parameter(Position = 1, Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "The time grain of the query")]
+        [Parameter(ParameterSetName = GetAzureRmAMetricFullParamGroup, Position = 1, Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "The aggregation type of the query")]
         [ValidateNotNullOrEmpty]
-        public TimeSpan TimeGrain { get; set; }
```

Comment:
Metric is a special case.  The whole API is new. I cannot "not break it" in this case.
What would be the procedure to introduce breaking changes?